src/sage/rings/series_valuation.py

```diff
@@ -0,0 +1,75 @@
+"""
+Valuation for power series and Laurent series.
+"""
+
+from sage.rings.valuation.valuation import DiscreteValuation
+
+class SeriesValuation(DiscreteValuation):
+    r"""
+    A class for valuation on power series and Laurent Series
+
+    TESTS::
+
+        sage: K.<t> = LaurentSeriesRing(QQ)
+        sage: v = K.valuation()
+        sage: TestSuite(v).run()
+    """
+    def _repr_(self):
+        r"""
+        Return a string representation of this valuation.
+
+        EXAMPLES::
+
+            sage: K.<t> = LaurentSeriesRing(QQ)
+            sage: v = K.valuation()
+            sage: v  # indirect doctest
+            t-adic valuation
+        """
+        return "%s-adic valuation" % self.domain().uniformizer()
```

Comment:
```suggestion
        return "%s-adic valuation" % self.uniformizer()
```

I think it would be good to implement `uniformizer()` on the level of valuation.

Reply:
Actually, once the "category" test above passes, more tests coming from the category of discrete valuations are going to fail, namely that you are missing a bunch of methods: `uniformizer`, `reduce`, `lift`, `residue_field`.